cub/util_type.cuh

```diff
@@ -273,9 +274,47 @@ struct NullType
 template <int A>
 struct Int2Type
 {
-   enum {VALUE = A};
+    enum {VALUE = A};
 };
 
+/**
+ * \brief Allows algorithms that take a value as input to take a future value that is not computed yet at launch time.
+ *
+ * Note that it is user's responsibility to ensure that the result will be ready before use via external synchronization
+ * or stream-ordering dependencies.
```

Comment:
Can you add an example `@code`/`@endcode` block showing how this is used?

Reply:
added